src/main/resources/org/jenkinsci/plugins/authorizeproject/form/dropdownDescriptorSelector.jelly

```diff
@@ -0,0 +1,73 @@
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout"
+         xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+  <st:documentation>
+    Renders a single &lt;select> control for choosing a Describable.
+    Depending on the currently selected value, its config.jelly will be
+    rendered below &lt;select, allowing the user to configure Describable.
+
+    <st:attribute name="field" use="required">
+      form field name. Used for databinding.
+    </st:attribute>
+    <st:attribute name="title" use="required">
+      Human readable title of this control.
+    </st:attribute>
+    <st:attribute name="lazy">
+      If specified, the additional the variables to be captured. See the @capture of &lt;renderOnDemand> tag.
+    </st:attribute>
+    <st:attribute name="descriptors">
+      Collection that lists up all the valid candidate descriptors.
+      If unspecified, inferred from the type of the field.
+    </st:attribute>
+    <st:attribute name="default">
+      If specified, this will be chosen as the default value in case the current selection is null. The default can be
+      an specific instance or a descriptor e.g.
+      ${descriptor.defaultSettingsProvider} or ${descriptor.defaultSettingsProvider.descriptor}. In the later case, the
+      from input fields will be empty.
+    </st:attribute>
+  </st:documentation>
+  <f:prepareDatabinding/>
+  <j:set target="${attrs}" property="descriptors"
+         value="${attrs.descriptors ?: descriptor.getPropertyType(instance,attrs.field).getApplicableDescriptors()}"/>
+
+  <f:dropdownList name="${attrs.field}" title="${attrs.title}" help="${descriptor.getHelpFile(attrs.field)}">
+    <d:invokeBody/>
+
+    <j:set var="current" value="${instance[attrs.field]}"/>
+    <j:set var="current" value="${current!=null ? current : (default.descriptor!=null ? default : null)}"/>
+    <j:forEach var="descriptor" items="${attrs.descriptors}" varStatus="loop">
+      <f:dropdownListBlock value="${loop.index}" title="${descriptor.displayName}"
+                           selected="${current.descriptor==descriptor || (current==null and descriptor==attrs.default)}"
+                           staplerClass="${descriptor.clazz.name}"
+                           lazy="${attrs.lazy?attrs.lazy+',':''}descriptor,it">
+        <l:ajax>
+          <j:set var="instance" value="${current.descriptor==descriptor ? current : null}"/>
+          <st:include from="${descriptor}" page="${descriptor.configPage}"/>
+        </l:ajax>
+      </f:dropdownListBlock>
+    </j:forEach>
+  </f:dropdownList>
+</j:jelly>
```

Comment:
`lazy` attribute captures `it` and allows capturing additional vars... I'll be pushing that fix into core

Sent from my iPhone

> On 14 Jan 2016, at 23:41, ikedam <notifications@github.com> wrote:
> 
> In src/main/resources/org/jenkinsci/plugins/authorizeproject/form/dropdownDescriptorSelector.jelly:
> 
> > +
> > +    <j:set var="current" value="${instance[attrs.field]}"/>
> > +    <j:set var="current" value="${current!=null ? current : (default.descriptor!=null ? default : null)}"/>
> > +    <j:forEach var="descriptor" items="${attrs.descriptors}" varStatus="loop">
> > +      <f:dropdownListBlock value="${loop.index}" title="${descriptor.displayName}"
> > +                           selected="${current.descriptor==descriptor || (current==null and descriptor==attrs.default)}"
> > +                           staplerClass="${descriptor.clazz.name}"
> > +                           lazy="${attrs.lazy?attrs.lazy+',':''}descriptor,it">
> > +        <l:ajax>
> > +          <j:set var="instance" value="${current.descriptor==descriptor ? current : null}"/>
> > +          <st:include from="${descriptor}" page="${descriptor.configPage}"/>
> > +        </l:ajax>
> > +      </f:dropdownListBlock>
> > +    </j:forEach>
> > +  </f:dropdownList>
> > +</j:jelly>
> What's the difference from the standard dropdownDescriptorSelector?
> 
> —
> Reply to this email directly or view it on GitHub.
> 
> -- 
> You received this message because you are subscribed to the Google Groups "engineering-code-reviews" group.
> To unsubscribe from this group and stop receiving emails from it, send an email to engineering-code-reviews+unsubscribe@cloudbees.com.
> To post to this group, send email to engineering-code-reviews@cloudbees.com.
> To view this discussion on the web visit https://groups.google.com/a/cloudbees.com/d/msgid/engineering-code-reviews/jenkinsci/authorize-project-plugin/pull/13/r49803067%40github.com.